backend/app/infrastructure/database/score_delegation.py

```diff
@@ -0,0 +1,23 @@
+from app.infrastructure.database.models import ScoreDelegation
+
+
+def does_delegation_exist(hashed_addr: str) -> bool:
+    delegation = ScoreDelegation.query.filter_by(hashed_addr=hashed_addr).first()
+
+    return delegation is not None
+
+
+def save_delegation(
+    primary_addr_hash: str, secondary_addr_hash: str, both_addr_hash: str
+):
+    delegation_primary = ScoreDelegation(hashed_addr=primary_addr_hash)
+    delegation_secondary = ScoreDelegation(hashed_addr=secondary_addr_hash)
+    delegation_both = ScoreDelegation(hashed_addr=both_addr_hash)
+
+    delegation_primary.save()
```

Comment:
i guess we want to save it atomically so maybe using .add() and .commit() as it is now instead of .save() (that does .add() and .commit() on one go) will be a better choice?